### PR TITLE
Migrate to charm.v6-unstable and macaroon-bakery.v1.

### PR DESCRIPTION
**Pull request by rogpeppe**
This is exactly https://github.com/juju/juju/pull/3228 re-proposed
with dependencies updated so that the tests should pass on windows
and so that no dependencies use the apparently problematic encoding
package.

(Review request: http://reviews.vapour.ws/r/2655/)

### CODE REVIEW

**Comment by urosj** [comment]
:+1:

**Comment by fabricematrat** [comment]
:+1:

File: dependencies.tsv [hunk 1]
```diff
@@ -38,12 +39,14 @@ gopkg.in/amz.v3	git	bff3a097c4108da57bb8cbe3aad2990d74d23676	2015-08-20T12:28:33
...
+gopkg.in/juju/charm.v6-unstable	git	49f6312f83592ab73309a9de5987d49a789c806c	2015-09-02T21:57:54Z
+gopkg.in/juju/charmrepo.v1	git	8677b12d9772a2a596a99e65b7e6f642fceb6c40	2015-09-14T13:26:00Z
+gopkg.in/juju/charmstore.v5-unstable	git	a921c6c69d74361c38a99a95d2aceec76533038d	2015-08-27T20:19:40Z
 gopkg.in/juju/environschema.v1	git	16cc59268c09c22870cb4de8eb6248652535f315	2015-08-24T13:22:26Z
-gopkg.in/macaroon-bakery.v0	git	9593b80b01ba04b519769d045dffd6abd827d2fd	2015-04-10T07:46:55Z
+gopkg.in/juju/jujusvg.v1	git	3eedb1c722ece2b66d62508368ca3e8b7f916569	2015-05-20T11:48:32Z
+gopkg.in/macaroon-bakery.v1	git	4ef12b7e1bf76f499808a230a099019fe9b7f192	2015-09-03T21:38:03Z
 gopkg.in/macaroon.v1	git	ab3940c6c16510a850e1c2dd628b919f0f3f1464	2015-01-21T11:42:31Z
-gopkg.in/mgo.v2	git	3569c88678d88179dcbd68d02ab081cbca3cd4d0	2015-06-04T15:26:27Z
+gopkg.in/mgo.v2	git	f4923a569136442e900b8cf5c1a706c0a8b0883c	2015-08-21T15:31:23Z
```
Comment (cmars) [inline dependencies.tsv]: What are the changes in mgo that are needed for this PR?

File: dependencies.tsv [hunk 2]
```diff
@@ -51,4 +54,4 @@ launchpad.net/gnuflag	bzr	roger.peppe@canonical.com-20140716064605-pk32dnmfust02
 launchpad.net/golxc	bzr	ian.booth@canonical.com-20141121040613-ztm1q0iy9rune3zt	13
 launchpad.net/gomaasapi	bzr	michael.foord@canonical.com-20150703101140-oo7493pkzlzg7l6u	63
 launchpad.net/gwacl	bzr	andrew.wilkins@canonical.com-20150811023840-kvosbu9d0kwwjfm2	245
-launchpad.net/tomb	bzr	gustavo@niemeyer.net-20130531003818-70ikdgklbxopn8x4	17
+launchpad.net/tomb	bzr	gustavo@niemeyer.net-20140529072043-hzcrlnl3ygvg914q	18
```
Comment (cmars) [inline dependencies.tsv]: What are the changes in tomb that are needed for this PR?

**Comment by cmars** [comment]
I'm a little concerned with the newer versions of mgo & tomb, since changes in those dependencies have a lot of potential for introducing side-effects elsewhere in core.

**Comment by rogpeppe** [comment]
I've reverted the mgo and tomb deps changes.

**Comment by rogpeppe** [comment]
$$merge$$

**Comment by rogpeppe** [comment]
This has now been tested against CI as a feature branch and has been blessed.

**Comment by axw** [comment]
LGTM

**Comment by rogpeppe** [comment]
$$merge$$

**Comment by rogpeppe** [comment]
$$merge$$

**Comment by rogpeppe** [comment]
$$merge$$

**Pull request merged**